### ISSUE TITLE
modifications should be copied with strand

### ISSUE THREAD

**Issue by dave-doty** [UC-Davis-molecular-computing/scadnano#745]
Create a strand with a modification:

![image](https://user-images.githubusercontent.com/19274365/152145201-432914f6-bd46-46c7-ba8f-98841b98015b.png)

Copy and paste it. The pasted strand lacks the modification, but should have it:

![image](https://user-images.githubusercontent.com/19274365/152145221-47f44faf-186b-482e-a65b-2d649853da22.png)

**Comment by dave-doty** [comment]
This is somewhat tricky when copying from one design to another. This is because modification data is not stored directly in the strand JSON, only the `id` field that is used as a key to access the full Modification data elsewhere in the Design JSON.

For example, here is what is copied to the clipboard after selecting the first strand above:

```json
{
  "strands": [
    {
      "color": "#32b86c",
      "domains": [
        {"helix": 6, "forward": true, "start": 10, "end": 31}
      ],
      "5prime_modification": "5'-Q"
    }
  ],
  "helices_view_order": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
}
```

In the Dart object, the full modification *is* stored in the Strand object, in this case as `Strand.modification_5p`, but only the `id` is written when serializing to JSON:

```dart
if (this.modification_5p != null) {
  json_map[constants.modification_5p_key] = this.modification_5p.id;
}
```

Probably the best way to handle this is the following. When copying Strand data to the clipboard, store any modifications as a separate top-level field in the JSON, under the key "modifications_in_design" (just like when writing a `.sc` file). Only store each modification once, just like with that scheme, i.e., if 2 different strands both have a modification with the same `id`, only store one copy of the modification. The storage should be just like the `.sc` file, for instance

```json
{
  "strands": [
    {
      "color": "#32b86c",
      "domains": [
        {"helix": 6, "forward": true, "start": 10, "end": 31}
      ],
      "5prime_modification": "5'-Q"
    }
  ],
  "helices_view_order": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
  "modifications_in_design": {
    "5'-Q": {
      "display_text": "Q",
      "idt_text": "Q",
      "display_connector": true,
      "location": "5'"
    }
  }
}
```

**Comment by dave-doty** [comment]
Closed by #823
